bvm/ballerina-core/src/main/java/org/ballerinalang/util/TableIterator.java

```diff
@@ -0,0 +1,325 @@
+            if (conn != null && !conn.isClosed()) {
+                conn.close();
+            }
+        } catch (SQLException e) {
+            throw new BallerinaException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public String getString(int columnIndex) {
+        try {
+            return rs.getString(columnIndex);
+        } catch (SQLException e) {
+            throw new BallerinaException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public long getInt(int columnIndex) {
+        try {
+            return rs.getLong(columnIndex);
+        } catch (SQLException e) {
+            throw new BallerinaException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public double getFloat(int columnIndex) {
+        try {
+            return rs.getDouble(columnIndex);
+        } catch (SQLException e) {
+            throw new BallerinaException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public boolean getBoolean(int columnIndex) {
+        try {
+            return rs.getBoolean(columnIndex);
+        } catch (SQLException e) {
+            throw new BallerinaException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public String getBlob(int columnIndex) {
+        try {
+            Blob bValue = rs.getBlob(columnIndex);
+            byte[] bdata = bValue.getBytes(1, (int) bValue.length());
+            return new String(bdata);
+        } catch (SQLException e) {
+            throw new BallerinaException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public Object[] getStruct(int columnIndex) {
+        Object[] objArray = null;
+        try {
+            Struct data = (Struct) rs.getObject(columnIndex);
+            if (data != null) {
+                objArray = data.getAttributes();
+            }
+        } catch (SQLException e) {
+            throw new BallerinaException(e.getMessage(), e);
+        }
+        return objArray;
+    }
+
+    @Override
+    public Object[] getArray(int columnIndex) {
+        try {
+            return generateArrayDataResult(rs.getArray(columnIndex));
+        } catch (SQLException e) {
+            throw new BallerinaException(e.getMessage(), e);
+        }
+    }
+
+    private Object[] generateArrayDataResult(Array array) throws SQLException {
+        Object[] objArray = null;
+        if (!rs.wasNull()) {
+            objArray = (Object[]) array.getArray();
+        }
+        return objArray;
+    }
+
+    @Override
+    public BStruct generateNext() {
+        BStruct bStruct = new BStruct(type);
+        int longRegIndex = -1;
+        int doubleRegIndex = -1;
+        int stringRegIndex = -1;
+        int booleanRegIndex = -1;
+        int refRegIndex = -1;
+        int blobRegIndex = -1;
+        int index = 0;
+        BStructType.StructField[] structFields = type.getStructFields();
+        for (BStructType.StructField sf : structFields) {
+            BType type = sf.getFieldType();
+            try {
```

Comment:
shall we wrap the for loop within try catch?(instead of having try catch inside the for loop)